h/views/api/groups.py

```diff
@@ -64,12 +66,29 @@ def create(request):
                 _("group with groupid '{}' already exists").format(groupid)
             )
 
-    group = group_create_service.create_private_group(
-        name=appstruct["name"],
-        userid=request.user.userid,
-        description=appstruct.get("description", None),
-        groupid=groupid,
-    )
+    group_type = appstruct.get("type", DEFAULT_GROUP_TYPE)
+
+    kwargs = {
+        "name": appstruct["name"],
+        "userid": request.user.userid,
+        "description": appstruct.get("description", None),
+        "groupid": groupid,
+    }
+
+    if group_type == "private":
+        method = group_create_service.create_private_group
+    else:
+        assert group_type in ("restricted", "open")
```

Comment:
Doesn't `.validate(payload)` take care of this?

Reply:
It does, the `assert` here is more for documentation purposes, the same as a comment `# Since group_type is not "private" then it must be either "restricted" or "open"` but the `assert` is more concise and will actually raise `AssertionError` if it fails